napari/_vispy/layers/labels.py

```diff
@@ -42,27 +42,25 @@
 
 ColorTuple = Tuple[float, float, float, float]
 
-low_disc_lookup_shader = """
-uniform sampler2D texture2D_LUT;
+if TYPE_CHECKING:
+    from napari.layers import Labels
 
-vec4 sample_label_color(float t) {
-    float phi_mod = 0.6180339887498948482;  // phi - 1
-    float value = 0.0;
-    float margin = 1.0 / 256;
 
+auto_lookup_shader = """
+uniform sampler2D texture2D_values;
+
+vec4 sample_label_color(float t) {
     if (t == 0) {
         return vec4(0);
     }
 
     if (($use_selection) && ($selection != t)) {
         return vec4(0);
     }
-
-    value = mod((t * phi_mod + $seed), 1.0) * (1 - 2*margin) + margin;
-
+    t = mod(t, $color_map_size - 1);
     return texture2D(
-        texture2D_LUT,
-        vec2(0.0, clamp(value, 0.0, 1.0))
+        texture2D_values,
+        vec2(0.0, (t + 1.5) / $color_map_size)
```

Comment:
why +1.5? Explain in a comment in the shader, ideally.

Reply:
fixed in different way